practitioner-service/src/main/java/uk/ac/ox/ndph/mts/practitioner_service/repository/FhirContextWrapper.java

```diff
@@ -17,41 +18,62 @@
 public class FhirContextWrapper {
     private final FhirContext fhirContext;
 
+    @SuppressWarnings("FieldMayBeFinal") // Value injection shouldn't be final
+    @Value("${fhir.uri}")
+    private String fhirUri = "";
+
     /**
      * constructor
-     */ 
+     */
     public FhirContextWrapper() {
         fhirContext = FhirContext.forR4();
     }
-    
+
     /**
      * Pretty print a resource
+     *
      * @param resource the resource to print
-     * @return the pretty print String of the resource 
+     * @return the pretty print String of the resource
      */
     public String prettyPrint(IBaseResource resource) {
-        return  fhirContext.newJsonParser().setPrettyPrint(true).encodeResourceToString(resource);
+        return fhirContext.newJsonParser().setPrettyPrint(true).encodeResourceToString(resource);
     }
 
     /**
      * Execute a bundle transaction to a FHIR endpoint
-     * @param uri the FHIR endpoint URI
+     *
      * @param input the Bundle to save
      * @return the returned Bundle object from FHIR endpoint
      */
-    public Bundle executeTrasaction(String uri, Bundle input) {
-        return fhirContext.newRestfulGenericClient(uri).transaction()
-                    .withBundle(input).execute();
+    public Bundle executeTransaction(Bundle input) {
+        return fhirContext.newRestfulGenericClient(fhirUri).transaction()
```

Comment:
following our call, wouldn't it will be better to have RestfulGenericClient initialized inside the ctor so there will be only one instance per scope of FhirContextWrapper?

Reply:
I would like to do this, but in a separate task/PR since too many hands are working on this service currently. Please resolve if you agree.